### PR TITLE
Search UI for discharge patients

### LINKED ISSUE #9312: Update search UI in discharge patients page
**Is your feature request related to a problem? Please describe.**
Update search UI for Discharge Patients page like it was it was implemented in patients index page.

**Describe the solution you'd like**
Replace existing search UI in discharge patients page with smart search UI

Before
Image

After
Image

### PR DESCRIPTION
**Pull request by Mahendar0701**
## Proposed Changes

- Fixes #9312 
- Updated search UI for discharge patients

image

@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

## Summary by CodeRabbit

- **New Features**
	- Introduced enhanced search functionality for discharged patients, allowing searches by multiple criteria such as phone number, name, and patient number.
	- Added a loading state to improve user experience during searches.
	- Replaced previous search input fields with a new `SearchByMultipleFields` component.

- **Bug Fixes**
	- Removed error states for phone number inputs to streamline the user interface.

- **Improvements**
	- Updated patient count display to reflect loading status accurately.

### CODE REVIEW

File: src/components/Facility/DischargedPatientsList.tsx [hunk 1]
```diff
@@ -52,16 +51,81 @@ const DischargedPatientsList = ({
...
+            ? value.length >= 13 || value === ""
+              ? value
+              : undefined
+            : undefined,
+      };
+
+      updateQuery(updatedQuery);
+    },
+    [updateQuery],
+  );
```
[the remark this answers is not in the document]
Comment (Mahendar0701) [reply src/components/Facility/DischargedPatientsList.tsx]: Done

**Comment by Mahendar0701** [comment]
@coderabbitai full review

File: src/components/Facility/DischargedPatientsList.tsx [hunk 2]
```diff
@@ -200,56 +262,12 @@ const DischargedPatientsList = ({
...
-      className: "grow w-full mb-2",
-    };
-  };
   const [diagnoses, setDiagnoses] = useState<ICD11DiagnosisModel[]>([]);
   const [phone_number, setPhoneNumber] = useState("");
-  const [phoneNumberError, setPhoneNumberError] = useState("");
   const [emergency_phone_number, setEmergencyPhoneNumber] = useState("");
-  const [emergencyPhoneNumberError, setEmergencyPhoneNumberError] =
-    useState("");
+  useState("");
```
[the remark this answers is not in the document]
Comment (Mahendar0701) [reply src/components/Facility/DischargedPatientsList.tsx]: Done

**Comment by Mahendar0701** [comment]
Reordered Discharged patient search buttons as implemented in PR [#9319](https://github.com/ohcnetwork/care_fe/pull/9319)

**Comment by Mahendar0701** [comment]
Hey @nihal467  can you add needs review and needs testing label for this PR

**Comment by Mahendar0701** [comment]
Hey @Jacobjeevan can you review this PR. Is this relevant PR or should i need to change anything.

File: src/components/Facility/DischargedPatientsList.tsx [hunk 3]
```diff
@@ -42,6 +39,8 @@ import routes from "@/Utils/request/api";
 import useQuery from "@/Utils/request/useQuery";
 import { formatPatientAge, humanizeStrings } from "@/Utils/utils";
 
+import SearchByMultipleFields from "../Common/SearchByMultipleFields";
```
Comment (Jacobjeevan) [inline src/components/Facility/DischargedPatientsList.tsx]: Absolute import. Other than that, LGTM.

**Review by Jacobjeevan — APPROVED** [review]

File: src/components/Facility/DischargedPatientsList.tsx (hunk 3 above)
Comment (Mahendar0701) [reply src/components/Facility/DischargedPatientsList.tsx]: Done. added

**Comment by Mahendar0701** [comment]
Hey @rithviknishad can you share your review. Let me know if any changes are required.

File: src/components/Facility/DischargedPatientsList.tsx [hunk 4]
```diff
@@ -518,3 +506,5 @@ const PatientListItem = ({ patient }: { patient: PatientModel }) => {
     </div>
   );
 };
+
+//
```
Comment (rithviknishad) [inline src/components/Facility/DischargedPatientsList.tsx]:
```suggestion
```

**Review by rithviknishad — APPROVED** [review]

**Comment by nihal467** [comment]
LGTM

**Review by nihal467 — APPROVED** [review]

**Comment by rithviknishad** [comment]
@shivankacker Would this page be overwritten/removed in your branch?

**Comment by Mahendar0701** [comment]
Hey @rithviknishad Can you please assign story points for this issue

**Comment by shivankacker** [comment]
@rithviknishad no, the discharge patients page is not affected in my PR

**Pull request merged**